Source/santasyncservice/SNTSyncEventUpload.mm

```diff
@@ -0,0 +1,183 @@
+/// Copyright 2015 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+#import "Source/santasyncservice/SNTSyncEventUpload.h"
+
+#import <MOLCertificate/MOLCertificate.h>
+#import <MOLXPCConnection/MOLXPCConnection.h>
+
+#import "Source/common/SNTCommonEnums.h"
+#import "Source/common/SNTConfigurator.h"
+#import "Source/common/SNTFileInfo.h"
+#import "Source/common/SNTLogging.h"
+#import "Source/common/SNTStoredEvent.h"
+#import "Source/common/SNTSyncConstants.h"
+#import "Source/common/SNTXPCControlInterface.h"
+#import "Source/common/String.h"
+#import "Source/santasyncservice/NSData+Zlib.h"
+#import "Source/santasyncservice/SNTSyncLogging.h"
+#import "Source/santasyncservice/SNTSyncState.h"
+
+#include <google/protobuf/arena.h>
+#include "Source/santasyncservice/syncv1.pb.h"
+namespace pbv1 = ::santa::sync::v1;
+
+using santa::common::NSStringToUTF8String;
+
+@implementation SNTSyncEventUpload
+
+- (NSURL *)stageURL {
+  NSString *stageName = [@"eventupload" stringByAppendingFormat:@"/%@", self.syncState.machineID];
+  return [NSURL URLWithString:stageName relativeToURL:self.syncState.syncBaseURL];
+}
+
+- (BOOL)sync {
+  dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+  [[self.daemonConn remoteObjectProxy] databaseEventsPending:^(NSArray *events) {
+    if (events.count) {
+      [self uploadEvents:events];
+    }
+    dispatch_semaphore_signal(sema);
+  }];
+  return (dispatch_semaphore_wait(sema, DISPATCH_TIME_FOREVER) == 0);
+}
+
+- (BOOL)uploadEvents:(NSArray *)events {
+  google::protobuf::Arena arena;
+  auto req = google::protobuf::Arena::Create<::pbv1::EventUploadRequest>(&arena);
+  auto uploadEvents = req->mutable_events();
```

Comment:
Avoid `auto` where it isn't readily apparent what the type is.

https://google.github.io/styleguide/cppguide.html#Type_deduction

I would argue that we should not use `auto` for these getters. I would like to see what types are coming out them, especially during review.

Reply:
Yeah I agree.